src/main/java/org/traccar/protocol/SuntechProtocolDecoder.java

```diff
@@ -439,7 +441,9 @@ private Position decode2356(
                 position.set(Position.KEY_RPM, Integer.parseInt(values[index++]));
             }
 
-            if (values.length - index >= 2) {
+            int need = hbm == 1 ? 2 : 7;
```

Comment:
I like the approach, but I think we need a better name for this variable. Or maybe just inline it to avoid naming.